### PR TITLE
Feature/sc 109233/implement the ability to remove a card from

### PR DESCRIPTION
**Pull request by markhornak**
### Overview

**Shortcut**: [109233](https://app.shortcut.com/guru/story/109233/implement-the-ability-to-remove-a-card-from-a-folder)

this provides the ability to remove a card from a folder.

- added remove_card_from_folder method in `core.py`
- added remove_card helper method for Folder object in `data_objects.py'
- changed internal `.items` array to `__items`, updated `data_objects.py` appropriately. this was done to hide the internal items array and provided a method to return the __items array same pattern as __cards and __folders.

### CODE REVIEW

File: guru/core.py [hunk 1]
```diff
@@ -2450,6 +2450,57 @@ def add_folder(self, title, collection, parentFolder=None, description=None):
...
+
+    # grab the folder's slug, used in the URL below
+    folder_slug = clean_slug(folder_obj.slug)
+
+    # grab the card from the Folder, if we don't find it, error out
+    card_item_obj = find_by_name_or_id(folder_obj.cards, card_id)
+    if not card_item_obj:
+      raise ValueError(f"couldn't find card: {card} in the folder: {folder}")
+
+    # we have a card in the folder, get it's item_id
```
Comment (jahlbornG) [inline guru/core.py:2482]: nit, `it's` -> `its`

File: guru/data_objects.py [hunk 2]
```diff
@@ -97,10 +97,8 @@ def __init__(self, data, folder_items=[], guru=None, parent_folder=None):
     else:
       self.collection = None
 
-  # arrays to hold contents of the Folder.  Note: items array is exposed for other methods to add objects
-  # such as cards or folders to the Folders object. e.g. add_card_to_folder() updates the items array
-  # and will do a save_folder() call.
-    self.items = []
+  # internal arrays to hold contents of the Folder.
```
Comment (jahlbornG) [inline guru/data_objects.py:100]: nit, indentation on this comment looks off

File: guru/core.py [hunk 3]
```diff
@@ -2450,6 +2450,57 @@ def add_folder(self, title, collection, parentFolder=None, description=None):
...
       return Folder(response.json(), guru=self)
 
+  def remove_card_from_folder(self, card, folder):
+    """
+    Removes the card from the folder.
+    Args:
+      card (str, required): The ID or Card object to be removed.
+      folder (str, required): the ID or Folder Object from which to remove the card.
+    Returns:
+      None
```
Comment (jahlbornG) [inline guru/core.py:2460]: this return value is not correct

File: guru/core.py (hunk 1 above)
Comment (markhornak) [reply guru/core.py]: fixed here : [6033d2e](https://github.com/guruhq/py-sdk/pull/57/commits/6033d2ea16993af6257e7da1743a020a3ee43ea9)

File: guru/data_objects.py (hunk 2 above)
Comment (markhornak) [reply guru/data_objects.py]: fixed here : [6033d2e](https://github.com/guruhq/py-sdk/pull/57/commits/6033d2ea16993af6257e7da1743a020a3ee43ea9)

File: guru/core.py (hunk 3 above)
Comment (markhornak) [reply guru/core.py]: fixed here : [6033d2e](https://github.com/guruhq/py-sdk/pull/57/commits/6033d2ea16993af6257e7da1743a020a3ee43ea9)

**Review by jahlbornG — APPROVED** [review]

**Pull request merged**